### ISSUE TITLE
Can no longer override workflow/layout structures from a plugin

### ISSUE THREAD

**Issue by zenweasel** [reactioncommerce/reaction#1969]
### Expected behavior
Able to override an existing workflow/layout combo by adding a `layout` record in the registry.

### Actual Behavior
Layout record is written to the Shops collection, but the original record is always selected because it's first in the collection.

### Steps to Reproduce the Behavior
1. Clone the example-plug into `import/plugins/custom`
1. `reaction reset`
1. Observe that the new layout/workflow structure is not used.

I believe at one point there was a `reverse` on Shops.layout` so that custom layouts got loaded first but that appears to be gone and I have a recollection of trying that again and it broke other things.

We probably need a way to "merge" layout records where custom ones just take the place of the stock ones.

**Comment by gepolv** [comment]
There might be another reason causing this issue.

Inside this function:
```
export function ReactionLayout(options = {}) {
   const layout = options.layout || Session.get("DEFAULT_LAYOUT") || "coreLayout";
   const workflow = options.workflow || Session.get("DEFAULT_WORKFLOW") || "coreWorkflow";
```
I found "options.workflow" already had a value which was not "coreWorkflow". Then later on, it caused this line to pick up a wrong layout:
`shop.layout.find((x) => selectLayout(x, layout, workflow)) `

**Comment by impactmass** [comment]
I was able to reproduce this by the steps listed. 
Next steps is looking at possible solutions starting with the hints added on the issue, while I also need to go through the docs to get more understanding on layouts and workflows. More update later..

**Comment by gepolv** [comment]
I have a pull request to fix this issue, please take a look:
https://github.com/reactioncommerce/reaction/pull/2003

**Comment by aaronjudd** [comment]
@zenweasel now that priority field has been added to layouts, is this still an issue?

**Comment by zenweasel** [comment]
This was fixed via #2023. Now there is a problem the example plugin product grid, but that's a separate issue.
